### PR TITLE
OAS-5749 | Add notebook support 

### PR DESCRIPTION
**Pull request by forgems**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,6 @@ update-modules:
 		$(shell zutano go mod replacements)
 	go get \
 		$(shell zutano go mod latest \
-			github.com/arangodb-managed/apis \
+			github.com/arangodb-managed/apis@OAS-5677 \
```
Comment (lakhansamani) [inline Makefile:61]: Don't forget

Comment (Robert-Stam) [inline Makefile:61]: don't forget

File: cmd/notebook/create_notebook.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+				OrganizationID string
+				ProjectID      string
+				DeploymetnID   string
+				Name           string
+				Description    string
+				DiskSize       int32
+				NotebookModel  string
+			}{}
+
+			f.StringVarP(&cargs.OrganizationID, "organization-id", "o", cmd.DefaultOrganization(), "Identifier of the organization to create the deployment in")
```
Comment (Robert-Stam) [inline cmd/notebook/create_notebook.go:55]: deployment --> notebook

File: cmd/notebook/create_notebook.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			f.StringVarP(&cargs.ProjectID, "project-id", "p", cmd.DefaultProject(), "Identifier of the project to create the deployment in")
```
Comment (Robert-Stam) [inline cmd/notebook/create_notebook.go:56]: deployment --> notebook

File: cmd/notebook/create_notebook.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+			f.StringVarP(&cargs.DeploymetnID, "deployment-id", "d", "", "Identifier of the deployment that the notebook has to run next to")
+			f.StringVarP(&cargs.NotebookModel, "notebook-model", "m", "", "Identifier of the notebook model that the notebook has to use")
+			f.StringVarP(&cargs.Name, "name", "n", "", "Name of the notebook")
+			f.StringVarP(&cargs.Description, "description", "", "", "Description of the notebook")
+			f.Int32VarP(&cargs.DiskSize, "disk-size", "s", 0, "Disk size that has to be attached to given notebook")
```
Comment (Robert-Stam) [inline cmd/notebook/create_notebook.go:61]: Disk size (in Gib)

File: cmd/notebook/delete_notebook.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+					log.Fatal().Err(err).Msg("Failed to delete a notebook")
+				}
+				notebook, err := notebookc.GetNotebook(ctx, &common.IDOptions{
+					Id: id,
+				})
+				if err != nil {
+					log.Fatal().Err(err).Msg("Failed to get a notebook")
+				}
+
+				fmt.Println(format.Notebook(notebook, cmd.RootArgs.Format))
```
Comment (Robert-Stam) [inline cmd/notebook/delete_notebook.go:72]: why this in delete?

File: cmd/notebook/create_notebook.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+						NotebookModelId: cargs.NotebookModel,
+						DiskSize:        cargs.DiskSize,
+					},
+				}
+				created, err := notebookc.CreateNotebook(ctx, &n)
+				if err != nil {
+					log.Fatal().Err(err).Msg("Failed to create notebook")
+				}
+				format.DisplaySuccess(cmd.RootArgs.Format)
+				fmt.Printf("%#v\n", created)
```
Comment (Robert-Stam) [inline cmd/notebook/create_notebook.go:92]: fmt.Println(format.Notebook(crreated, cmd.RootArgs.Format))

**Review by lakhansamani — CHANGES_REQUESTED** [review]
Minor comments,
Further looks good to me

File: cmd/notebook/list_notebooks.go [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		cmd.ListCmd,
+		&cobra.Command{
+			Use:   "notebooks",
+			Short: "List notebooks",
+		},
+		func(c *cobra.Command, f *flag.FlagSet) {
+			cargs := &struct {
+				OrganizationID string
+				ProjectID      string
+				DeploymetnID   string
```
Comment (Robert-Stam) [inline cmd/notebook/list_notebooks.go:49]: DeploymetnID --> DeploymentID

File: cmd/notebook/list_notebooks.go [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+				ctx := cmd.ContextWithToken()
+
+				selection.MustSelectDeployment(ctx, log, deploymentID, cargs.ProjectID, cargs.OrganizationID, datac, rmc)
+
+				options := &common.ListOptions{
+					Page:     0,
+					PageSize: 20,
+				}
+				var result []*notebook.Notebook
+				for {
```
Comment (Robert-Stam) [inline cmd/notebook/list_notebooks.go:79]: there was a forEach available, however this works as well.

File: pkg/format/notebook.go [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+	return formatList(opts, list, func(i int) []kv {
+		x := list[i]
+		return []kv{
+			{"id", x.GetId()},
+			{"name", x.GetName()},
+			{"description", x.GetDescription()},
+			{"url", x.GetUrl()},
+			{"endpoint", x.GetStatus().GetEndpoint()},
+			{"paused", formatBool(opts, x.GetIsPaused())},
+			{"created-at", formatTime(opts, x.GetCreatedAt())},
```
Comment (Robert-Stam) [inline pkg/format/notebook.go:56]: phase?

**Review by Robert-Stam — COMMENTED** [review]
See remarks

**Review by lakhansamani — APPROVED** [review]
LGTM

**Review by Robert-Stam — APPROVED** [review]
LGTM

**Pull request merged**